### ISSUE TITLE
Botania Ring of Magnetism Crash

### ISSUE THREAD

**Issue by Manbeartooth** [NillerMedDild/Enigmatica2Expert#1088]
**Modpack version**
1.60

**Can the issue be reproduced?**
Yes

**Logs**

https://pastebin.com/izuJ4AVb
https://pastebin.com/4FK4Wdt4

**Issue**
When I have the Flux network set to wireless charge items in my inventory and I try and pick up a Dimensional Receiver that has the Octadic capacitor installed it crashes the game with the above errors.  Basically it looks like the wireless charging is not supported on the enderIO dimensional receivers and when it tries to "cast" it on the block it hangs and eventually crashes out and you have to then remove that item from your bags or restore from world backup.

If this is a feature request, feel free to delete the template.

**Comment by NillerMedDild** [comment]
Yikes, I'll get this reported, thank you :)

**Comment by Manbeartooth** [comment]
After some further testing it appears that just placing the block in the world and then trying to either break the item with a pickaxe or with a wrench, it crashes out as well, pretty much same exact error as well.

**Comment by NillerMedDild** [comment]
Hmm, okay. May I have that report as well please?

**Comment by Manbeartooth** [comment]
Sure.
https://pastebin.com/LKuSiU9Q

**Comment by NillerMedDild** [comment]
Does the crash happen if you disable your Flux Network's wireless charging?

**Comment by Manbeartooth** [comment]
Yes, it does keep happening, even with all charging disabled in the base.  So I think my initial assumption with FN being part of the issue is probably wrong.  Sorry about that.  I thought it was part of the issue because it charged the item on my bar once and then froze up.  I have also tried removing Optifine, just in case it was part of the issue and it still as the same issue regardless.
Let me know if there is anything else you want me to try out.

**Comment by Manbeartooth** [comment]
So just did some more digging on the interwebs and found this: https://github.com/SleepyTrousers/EnderIO/issues/5086
Looks like the same exact issue and it actually has to do with Botania and the magnetic ring.  So there is a fix here: https://github.com/Vazkii/Botania/commit/1701732aaccb02279b64ceeef3d6a0c0409a3bf5

**Comment by Massel84** [comment]
I have the exact same issue but without wireless charging of Flux Network. But I have a Thermal Battery in my inventory which is charing all items of my inventory, maybe this is causing the same issue.

The game crashed whilst ticking player
Error: java.lang.ClassCastException: net.minecraft.item.ItemAir cannot be cast to crazypants.enderio.base.power.forge.item.IInternalPoweredItem

I broke the EnderIO Simple SAG Mill and during the brake animation the game crashed.
Now my game is broken and I am not able to play anymore :(

**Comment by Manbeartooth** [comment]
So the issue is with the Ring of Magnetism from botania, just confirmed that in the game.  If you remove your magnet ring and then break the block it works and lets you pick it up with no issues.

**Comment by Massel84** [comment]
> So the issue is with the Ring of Magnetism from botania, just confirmed that in the game. If you remove your magnet ring and then break the block it works and lets you pick it up with no issues.

hmm I also had the ring of magnetism put on. But how to fix it now? I am not able to login anymore, so I can't put off the ring.

**Comment by Manbeartooth** [comment]
If you are playing single player, then the game should have been auto backing up the world save and you can just restore the world back to before your issue.  For MP, I think you can reset your character inventory, or maybe the server admin has to do that, can't remember, been a while since I did a MP server admin.

**Comment by MirrorImageQS** [comment]
I realize that this issue has already been closed, but after encountering this yesterday myself, I found a mitigation for anyone whose world is corrupted by this pending a new full release.

In the Mod Config for Botania, there's an option to "Invert Magnet Ring Function", or something like that. In the botania.cfg file, it's "B:magnetRing.invert". Basically, it changes the functionality of the ring so that it only magnets while you are crouching, instead of the default behavior of disabling magneting while crouching. If you change that setting from false to true, it will allow you to load the world again.

The crash itself seems to happen when the ring starts to magnet an item. That does mean that you'd instead crash when you crouch though and the magnet turns on to try and suck up the item.
